File: code/library/src/main/java/org/fbme/lib/iec61499/stringify/ParameterAssignmentPrinter.java
package org.fbme.lib.iec61499.stringify;


import org.fbme.lib.iec61499.declarations.ParameterAssignment;
import org.jdom.Element;

import java.util.List;
import java.util.Objects;

public class ParameterAssignmentPrinter extends PrinterElementBase<ParameterAssignment> {

    public ParameterAssignmentPrinter(ParameterAssignment assignment) {
        super(assignment, "Parameter");
    }

    @Override
    protected void printElementBody(Element element) {
        element.setAttribute("Name", myElement.getParameterReference().getPresentation());
        element.setAttribute("Value", STPrinter.printLiteral(Objects.requireNonNull(myElement.getValue())));
    }

    public static void printAll(List<ParameterAssignment> assignments, Element element) {
        for (ParameterAssignment assignment : assignments) {
            element.addContent(new ParameterAssignmentPrinter(assignment).print());
        }
    }
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/stringify/ParameterAssignmentPrinter.kt
package org.fbme.lib.iec61499.stringify

import org.fbme.lib.iec61499.declarations.ParameterAssignment
import org.jdom.Element

class ParameterAssignmentPrinter(assignment: ParameterAssignment) :
    PrinterElementBase<ParameterAssignment>(assignment, "Parameter") {
    override fun printElementBody(element: Element) {
        element.setAttribute("Name", this.element.parameterReference.presentation)
        val value = this.element.value
        if (value != null) {
            element.setAttribute("Value", STPrinter.printLiteral(value))
        }
    }

    companion object {
        @JvmStatic
        fun printAll(assignments: List<ParameterAssignment>, element: Element) {
            for (assignment in assignments) {
                element.addContent(ParameterAssignmentPrinter(assignment).print())
            }
        }
    }
}
